File: server/src/main/java/com/king/hawkeye/server/dao/AlertConfigDao.java
package com.king.hawkeye.server.dao;

import com.king.hawkeye.server.entity.AlertConfig;
import org.apache.commons.lang.ArrayUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by King on 16/4/26.
 */
public class AlertConfigDao implements IAlertConfigDao {

    private static final Logger LOG = LogManager.getLogger(AlertConfigDao.class);

    private Connection connection = BaseDao.getConnection();

    @Override
    public List<AlertConfig> getByProjectId(int projectId) {
        List<AlertConfig> result = null;
        try {
            PreparedStatement prepareStatement = connection.prepareStatement("SELECT C.*,P.name as project_name FROM t_alert_config C left join t_project P on C.project_id = P.id where project_id = ?");
            prepareStatement.setInt(1, projectId);
            ResultSet rs = prepareStatement.executeQuery();
            int cols = rs.getMetaData().getColumnCount();
            result = new ArrayList<AlertConfig>(cols);
            while (rs.next()) {
                AlertConfig config = new AlertConfig();
                config.setId(rs.getInt("id"));
                config.setProjectId(rs.getInt("project_id"));
                config.setProjectName(rs.getString("project_name"));
                config.setConfig(rs.getString("config"));
                result.add(config);
            }
        } catch (SQLException e) {
            LOG.error("get all alert config error.", e);
            return null;
        }
        return result;
    }

    @Override
    public List<AlertConfig> getAll() {
        List<AlertConfig> result = null;
        try {
            PreparedStatement prepareStatement = connection.prepareStatement("SELECT C.*,P.name as project_name FROM t_alert_config C left join t_project P on C.project_id = P.id");
            ResultSet rs = prepareStatement.executeQuery();
            int cols = rs.getMetaData().getColumnCount();
            result = new ArrayList<AlertConfig>(cols);
            while (rs.next()) {
                AlertConfig config = new AlertConfig();
                config.setId(rs.getInt("id"));
                config.setProjectId(rs.getInt("project_id"));
                config.setProjectName(rs.getString("project_name"));
                config.setConfig(rs.getString("config"));
                result.add(config);
            }
        } catch (SQLException e) {
            LOG.error("get all alert config error.", e);
            return null;
        }
        return result;
    }

    @Override
    public Integer[] getIdsByProjectName(String projectName) {
        Integer[] result = new Integer[0];
        try {
            PreparedStatement preparedStatement = connection.prepareStatement("select id from t_project where name = ?");
            preparedStatement.setString(1, projectName);
            ResultSet rs = preparedStatement.executeQuery();
            if (rs.getMetaData().getColumnCount() <= 0) {
                return new Integer[0];
            }
            while (rs.next()) {
                int projectId = rs.getInt("id");

                List<AlertConfig> list = this.getByProjectId(projectId);
                for (AlertConfig config : list) {
                    result = (Integer[]) ArrayUtils.add(result, config.getId());
                }
            }
        } catch (SQLException e) {
            LOG.error("get config ids by project name error, project name is " + projectName, e);
        }
        return result;
    }
}

File: web/dao/alertConfigDao.js
/**
 * Created by King on 16/4/22.
 */
var db = require('./db');

var $sql = {
    queryByProjectId: 'SELECT C.id,C.project_id, P.name as project_name, C.config FROM t_alert_config C left join t_project P on C.project_id = P.id where project_id = ?',
    insert: 'INSERT INTO t_alert_config(project_id,config) VALUES (?,?)',
    deleteConfigById: 'delete from t_alert_config where id = ?'
}

module.exports = {
    queryByProjectId: function (res, projectId) {
        console.log("projectId : " + projectId);
        db.query($sql.queryByProjectId, [projectId], function (err, result) {
            if (err) {
                console.log('query alert config error. project id is ' + projectId);
                res.json({code: 500, msg: 'query alert config error.'});
            } else {
                // result.config = JSON.parse(result.config);
                res.json({code: 200, msg: result});
            }
        })
    },
    insert: function (res, project_id, config) {
        db.query($sql.insert, [project_id, config], function (err, result) {
            if(err) {
                console.log('insert alert config error. project_id = ' + project_id + ', config is ' + config + "\n" + err);
                res.json({code: 500, msg: 'insert alert config error.'});
            } else {
                res.json({code: 200, msg: result});
            }
        })
    },
    deleteConfigById: function (res, configId) {
        db.query($sql.deleteConfigById, [configId], function(err, result) {
            if(err) {
                console.log('delete alert config error. project_id = ' + project_id + ', config is ' + config + "\n" + err);
                res.json({code: 500, msg: 'insert alert config error.'});
            } else {
                console.log('delete alert config success: ' + result);
                res.json({code: 200, msg: result});
            }
        })
    }
}